src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -848,5 +1092,173 @@ public void Dispose()
                 result.TaskCompletionSource.TrySetCanceled();
             }
         }
+
+
+        public enum TxInstruction
+        {
+            BeginTx,
+            Commit,
+            Rollback,
+            DisposeTx,
+            FinishAsyncAndStartNew,
+
+            //Todo I think this is not relevant all the async instruction can run as one unit in  FinishAsyncAndStartNew
+            EndAsyncCommit
```

Comment:
We call `EndAsyncCommit` explicitly in some cases, though, no?

Reply:
done